packages/react-native-reanimated/Common/cpp/LayoutAnimations/LayoutAnimationsProxy.cpp

```diff
@@ -0,0 +1,754 @@
+#ifdef RCT_NEW_ARCH_ENABLED
+
+#include "LayoutAnimationsProxy.h"
+#include <react/renderer/animations/utils.h>
+#include <react/renderer/mounting/ShadowViewMutation.h>
+#include <set>
+#include <utility>
+#include "NativeReanimatedModule.h"
+
+namespace reanimated {
+
+// We never modify the Shadow Tree, we just send some additional
+// mutations to the mounting layer.
+// When animations finish, the Host Tree will represent the most recent Shadow
+// Tree
+std::optional<MountingTransaction> LayoutAnimationsProxy::pullTransaction(
+    SurfaceId surfaceId,
+    MountingTransaction::Number transactionNumber,
+    const TransactionTelemetry &telemetry,
+    ShadowViewMutationList mutations) const {
+#ifdef LAYOUT_ANIMATIONS_LOGS
+  LOG(INFO) << "\npullTransaction " << std::this_thread::get_id() << " "
+            << surfaceId << std::endl;
+#endif
+  auto lock = std::unique_lock<std::recursive_mutex>(mutex);
```

Comment:
On android this function is called from the JS thread. The entire flow there is a bit different, because `ShadowViewMutations` are transferred into `MountItems`, which are then passed to the platform.